packages/ui/nodes/table/src/PlateTableCellElement.tsx

```diff
@@ -18,61 +18,67 @@ export const getCssTableCellRoot = ({
   isHeader,
   selected,
   borders,
+  background,
 }: {
   hideBorder?: boolean;
   isHeader?: boolean;
   selected?: boolean;
   borders?: BorderStylesDefault;
-} = {}): CSSProp => [
-  tw`relative p-0 overflow-visible bg-white border-none`,
-  hideBorder && tw`before:border-none`,
-  !hideBorder && [
-    tw`before:content-[''] before:box-border before:absolute before:select-none`,
-    borders && [
-      borders.bottom &&
-        css`
-          ::before {
-            border-bottom: ${borders.bottom.size}px ${borders.bottom.style}
-              ${borders.bottom.color};
-          }
-        `,
-      borders.right &&
-        css`
-          ::before {
-            border-right: ${borders.right.size}px ${borders.right.style}
-              ${borders.right.color};
-          }
-        `,
-      borders.left &&
-        css`
-          ::before {
-            border-left: ${borders.left.size}px ${borders.left.style}
-              ${borders.left.color};
-          }
-        `,
-      borders.top &&
-        css`
-          ::before {
-            border-top: ${borders.top.size}px ${borders.top.style}
-              ${borders.top.color};
-          }
-        `,
+  background?: string;
+} = {}): CSSProp => {
+  const appliedBackground =
+    background || (isHeader && 'rgb(244, 245, 247)') || '';
+  return [
+    tw`relative p-0 overflow-visible bg-white border-none`,
+    hideBorder && tw`before:border-none`,
+    !hideBorder && [
+      tw`before:content-[''] before:box-border before:absolute before:select-none`,
+      borders && [
+        borders.bottom &&
+          css`
+            ::before {
+              border-bottom: ${borders.bottom.size}px ${borders.bottom.style}
+                ${borders.bottom.color};
+            }
+          `,
+        borders.right &&
+          css`
+            ::before {
+              border-right: ${borders.right.size}px ${borders.right.style}
+                ${borders.right.color};
+            }
+          `,
+        borders.left &&
+          css`
+            ::before {
+              border-left: ${borders.left.size}px ${borders.left.style}
+                ${borders.left.color};
+            }
+          `,
+        borders.top &&
+          css`
+            ::before {
+              border-top: ${borders.top.size}px ${borders.top.style}
+                ${borders.top.color};
+            }
+          `,
+      ],
     ],
-  ],
-  isHeader && tw`text-left`,
-  isHeader &&
-    css`
-      ::before {
-        background-color: rgb(244, 245, 247);
-      }
+    isHeader && tw`text-left`,
+    appliedBackground &&
+      css`
+        ::before {
+          background: ${appliedBackground};
```

Comment:
Can we use `background-color` to prevent some of the potential issues described [here](https://security.stackexchange.com/a/124858)? Same for anywhere that a user-specified background is applied in CSS.

Some validation at paste time would also be good, if you can figure out how to handle all the cases (rgb, rgba, hsv, hex, hex with alpha, etc.). Might be worth using an external library for CSS color parsing if you can find one.

Reply:
We're moving to tailwind very soon. How could we support this limitation?

https://tailwindcss.com/docs/content-configuration

![image](https://github.com/udecode/plate/assets/19695832/98e588ba-d556-473d-9122-e5693311b41d)

Reply:
Actually, I was just about to say that the interpolation here looks like it's vulnerable to injection attacks, `background-color` or no. Can you find a way to do this using a `style={{}}` prop?

Reply:
If not, we might be able to avoid use of `::before` altogether.

Reply:
The latter should work indeed.